docs/01_cva6_user/RISCV_Instructions_RVZcmp.rst

```diff
@@ -0,0 +1,169 @@
+
+    **Exception raised**: NONE
+
+- **CM.POP**: Compressed Pop
+
+    **Format**: cm.pop {reg_list}, stack_abj
+
+    **Description**: Destroy stack frame: load ra and 0 to 12 saved registers from the stack frame, deallocate the stack frame.
+
+    **Pseudocode**: if (XLEN==32) bytes=4; else bytes=8;
+                    addr=sp+stack_adj-bytes;
+                    for(i in 27,26,25,24,23,22,21,20,19,18,9,8,1) {
+                        if (xreg_list[i]) {
+                            switch(bytes) {
+                                4: asm("lw x[i], 0(addr)");
+                                8: asm("ld x[i], 0(addr)");
+                            }
+                            addr-=bytes;
+                        }
+                    }
+                    sp+=stack_adj;
+
+    **Invalid values**: reg_list not in {ra, s0-s11} and stack_adj not in {16, 32, 48, 64, 80, 96, 112, 128, 144, 160}.
+
+    **Exception raised**: NONE
+
+- **CM.POPRETZ**: Compressed Pop return zero
+
+    **Format**: cm.popretz {reg_list}, stack_adj
+
+    **Description**: Destroy stack frame: load ra and 0 to 12 saved registers from the stack frame, deallocate the stack frame, move zero into a0, return to ra.
+
+    **Pseudocode**: if (XLEN==32) bytes=4; else bytes=8;
+                    addr=sp+stack_adj-bytes;
+                    for(i in 27,26,25,24,23,22,21,20,19,18,9,8,1) {
+                        if (xreg_list[i]) {
+                            switch(bytes) {
+                                4: asm("lw x[i], 0(addr)");
+                                8: asm("ld x[i], 0(addr)");
+                            }
+                            addr-=bytes;
+                        }
+                    }
+                    asm("li a0, 0");
+                    sp+=stack_adj;
+                    asm("ret");
+
+    **Invalid values**: reg_list not in {ra, s0-s11} and stack_adj not in {16, 32, 48, 64, 80, 96, 112, 128, 144, 160}.
+
+    **Exception raised**: NONE
+
+- **CM.POPRET**: Compressed Pop return
+
+    **Format**: cm.popret {reg_list}, stack_adj
+
+    **Description**: Destroy stack frame: load ra and 0 to 12 saved registers from the stack frame, deallocate the stack frame, return to ra.
+
+    **Pseudocode**: if (XLEN==32) bytes=4; else bytes=8;
+                    addr=sp+stack_adj-bytes;
+                    for(i in 27,26,25,24,23,22,21,20,19,18,9,8,1) {
+                        if (xreg_list[i]) {
+                            switch(bytes) {
+                                4: asm("lw x[i], 0(addr)");
+                                8: asm("ld x[i], 0(addr)");
+                            }
+                            addr-=bytes;
+                        }
+                    }
+                    sp+=stack_adj;
+                    asm("ret");
+
+    **Invalid values**: reg_list not in {ra, s0-s11} and stack_adj not in {16, 32, 48, 64, 80, 96, 112, 128, 144, 160}.
```

Comment:
It is difficult to know what are the invalid values for `reg_list`. According to RISC-V spec, it is not only {ra, s0-s11} which is valid but also {ra}, {ra, s0}, {ra, s0-s1}, ..., {ra, s0-s9}.

Regarding stack_adj, the invalid values are depending on XLEN.

Reply:
Will something like the following make it easy to understand?
reg_list not in [ {ra}, {ra, s0}, {ra, s0-s1}, {ra, s0-s2}, ..., {ra, s0-s8}, {ra, s0-s9}, {ra, s0-s11} ], stack_adj not in [ 16, 32, 48, 64, 80, 96, 112 ] and [ 16, 32, 48, 64, 80, 96, 112, 128, 144, 160 ] for RV32 and RV64 respectively.

Reply:
> Will something like the following make it easy to understand? reg_list not in [ {ra}, {ra, s0}, {ra, s0-s1}, {ra, s0-s2}, ..., {ra, s0-s8}, {ra, s0-s9}, {ra, s0-s11} ], stack_adj not in [ 16, 32, 48, 64, 80, 96, 112 ] and [ 16, 32, 48, 64, 80, 96, 112, 128, 144, 160 ] for RV32 and RV64 respectively.

Yes, it is less ambiguous. Thanks.